liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,1401 @@
+    Balance, DepositData, DepositId, FarmId, FarmMultiplier, GlobalFarmData, GlobalFarmId, GlobalFarmState,
+    LoyaltyCurve, YieldFarmData, YieldFarmEntry, YieldFarmId, YieldFarmState,
+};
+use codec::{Decode, Encode, FullCodec};
+use frame_support::{
+    ensure,
+    pallet_prelude::*,
+    sp_runtime::{
+        traits::{AccountIdConversion, BlockNumberProvider, MaybeSerializeDeserialize, One, Zero},
+        RuntimeDebug,
+    },
+    PalletId,
+};
+
+use sp_runtime::ArithmeticError;
+
+use hydra_dx_math::liquidity_mining as math;
+use hydradx_traits::liquidity_mining::Handler;
+use orml_traits::MultiCurrency;
+use scale_info::TypeInfo;
+use sp_arithmetic::{
+    traits::{CheckedDiv, CheckedSub},
+    FixedPointNumber, FixedU128, Permill,
+};
+use sp_std::convert::{From, Into, TryInto};
+
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+type AssetIdOf<T> = <T as pallet::Config>::CurrencyId;
+type BlockNumberFor<T> = <T as frame_system::Config>::BlockNumber;
+type PeriodOf<T> = <T as frame_system::Config>::BlockNumber;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_system::pallet_prelude::BlockNumberFor;
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + TypeInfo {
+        /// Asset type.
+        type CurrencyId: Parameter + Member + Copy + MaybeSerializeDeserialize + Ord + From<u32>;
+
+        /// Currency for transfers.
+        type MultiCurrency: MultiCurrency<Self::AccountId, CurrencyId = Self::CurrencyId, Balance = Balance>;
+
+        /// Pallet id.
+        type PalletId: Get<PalletId>;
+
+        /// Minimum total rewards to distribute from global farm during liquidity mining.
+        #[pallet::constant]
+        type MinTotalFarmRewards: Get<Balance>;
+
+        /// Minimum number of periods to run liquidity mining program.
+        #[pallet::constant]
+        type MinPlannedYieldingPeriods: Get<Self::BlockNumber>;
+
+        /// Minimum user's deposit to start yield farming.
+        #[pallet::constant]
+        type MinDeposit: Get<Balance>;
+
+        /// The block number provider
+        type BlockNumberProvider: BlockNumberProvider<BlockNumber = Self::BlockNumber>;
+
+        /// Id used to identify amm pool in liquidity mining pallet.
+        type AmmPoolId: Parameter + Member + Clone + FullCodec;
+
+        type Handler: hydradx_traits::liquidity_mining::Handler<
+            Self::CurrencyId,
+            Self::AmmPoolId,
+            GlobalFarmId,
+            FarmId,
+            Balance,
+            DepositId,
+            Self::AccountId,
+        >;
+
+        /// Maximum number of yield farms same LP shares can be re/deposited into. This value always
+        /// MUST BE >= 1.         
+        #[pallet::constant]
+        type MaxFarmEntriesPerDeposit: Get<u8>;
```

Comment:
What if you use Range or RangeInclusive? 

that way, you can specify min and max bounds easily. You wont need to check if it > 1, you would just check if value is within range. 

```suggestion
        type MaxFarmEntriesPerDeposit: Get<RangeInclusive<u8>>;
```